tests/test_uncrustify/utilities.py

```diff
@@ -0,0 +1,193 @@
+# Logic for listing and running tests.
+#
+# * @author  Ben Gardner        October 2009
+# * @author  Guy Maurel         October 2015
+# * @author  Matthew Woehlke    June 2018
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+from .config import config, all_tests, FAIL_ATTRS, PASS_ATTRS, SKIP_ATTRS
+from .failure import Failure, MismatchFailure, UnstableFailure
+from .test import Test
+
+try:
+    from .ansicolor import printc
+except Exception:
+    def printc(ctext, ntext, *args, **kwargs):
+        print(ctext + ntext)
```

Comment:
I am not sure why the try catch is needed here ... also the dot before the file name in the from import is a bit strange to me, shouldn't ./ also work here?

Reply:
> I am not sure why the try catch is needed here

The way `ansicolor.py` is implemented is a little quirky... basically, it tries to do some global initialization that relies on curses being functional (e.g. `curses.setupterm()`). Having the module assume that curses works, and setting up the fallback this way, seemed less bizarre than having large chunks of `ansicolor.py` inside of a `try`. OTOH, I think I've convinced myself I can live with the alternative...

>  the dot before the file name in the from import is a bit strange to me, shouldn't ./ also work here?

Nope, `./` is a syntax error. `from .<name> import <...>` is the way to do relative imports. (Basically, it forces that `<name>` is a sibling of the file doing the importing, as opposed to searching the package paths.)